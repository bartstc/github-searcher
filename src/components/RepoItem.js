import React from 'react';
import { connect } from 'react-redux';
import { saveRepo } from '../modules/savedRepos/savedActions';

export const RepoItem = ({ repo, saveRepo }) => {
  const handleSave = () => saveRepo(repo);

  const { owner: { login }, html_url, full_name, description, language } = repo;

  return (
    <div className="card mt-4">
      <div className="card-header d-flex justify-content-between align-items-center">
        <div className="font-weight-bold">User: {login}</div>
        <button
          type="button"
          className="btn btn-warning"
          onClick={handleSave}
        >Save</button>
      </div>
      <div className="card-body">
        <a className="link-unstyled" href={html_url} target="_blank" rel="noopener noreferrer"><h5 className="card-title">{full_name}</h5></a>
        <p className="card-text">{description}</p>
        <span className="badge badge-warning badge-pill mr-3">{language}</span>
      </div>
    </div>
  );
};

export default connect(null, { saveRepo })(RepoItem);