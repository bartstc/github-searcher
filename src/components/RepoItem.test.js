import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { RepoItem } from './RepoItem';

describe('RepoItem', () => {
  const props = {
    repo: {
      owner: { login: 'test' },
      html_url: 'test',
      full_name: 'test',
      description: 'test',
      language: 'test'
    }
  };
  const item = shallow(<RepoItem {...props} />);

  it('renders correctly', () => {
    expect(toJson(item)).toMatchSnapshot();
  });
});