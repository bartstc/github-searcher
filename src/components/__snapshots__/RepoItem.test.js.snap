// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoItem renders correctly 1`] = `
<div
  className="card mt-4"
>
  <div
    className="card-header d-flex justify-content-between align-items-center"
  >
    <div
      className="font-weight-bold"
    >
      User: 
      test
    </div>
    <button
      className="btn btn-warning"
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>
  <div
    className="card-body"
  >
    <a
      className="link-unstyled"
      href="test"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h5
        className="card-title"
      >
        test
      </h5>
    </a>
    <p
      className="card-text"
    >
      test
    </p>
    <span
      className="badge badge-warning badge-pill mr-3"
    >
      test
    </span>
  </div>
</div>
`;
